services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -117,59 +130,105 @@ async def _get_service_access_rights(
 
 
 async def _create_services_in_db(
-    app: FastAPI,
+    connection: SAConnection,
     service_keys: Set[Tuple[ServiceKey, ServiceVersion]],
     services: Dict[Tuple[ServiceKey, ServiceVersion], ServiceDockerData],
 ) -> None:
-    engine: Engine = app.state.engine
-    async with engine.acquire() as conn:
-        services_repo = ServicesRepository(conn)
-        for service_key, service_version in service_keys:
-            service: ServiceDockerData = services[(service_key, service_version)]
-            # find the service owner
-            owner_gid, service_access_rights = await _get_service_access_rights(
-                service, conn
-            )
-            # set the service in the DB
-            await services_repo.create_service(
-                ServiceMetaDataAtDB(**service.dict(), owner=owner_gid),
-                service_access_rights,
-            )
+
+    services_repo = ServicesRepository(connection)
+    for service_key, service_version in service_keys:
+        service: ServiceDockerData = services[(service_key, service_version)]
+        # find the service owner
+        owner_gid, service_access_rights = await _get_service_access_rights(
+            service, connection
+        )
+        # set the service in the DB
+        await services_repo.create_service(
+            ServiceMetaDataAtDB(**service.dict(), owner=owner_gid),
+            service_access_rights,
+        )
+
+
+async def _ensure_services_metadata_uptodate(
+    app: FastAPI, connection: SAConnection
+) -> None:
+    services_in_registry: Dict[
+        Tuple[ServiceKey, ServiceVersion], ServiceDockerData
+    ] = await _list_registry_services(app)
+    services_in_db: Set[Tuple[ServiceKey, ServiceVersion]] = await _list_db_services(
+        connection
+    )
+
+    # check that the db has all the services at least once
+    missing_services_in_db = set(services_in_registry.keys()) - services_in_db
+    if missing_services_in_db:
+        logger.debug(
+            "missing services in db:\n%s", pformat(missing_services_in_db),
+        )
+        # update db (rationale: missing services are shared with everyone for now)
+        await _create_services_in_db(
+            connection, missing_services_in_db, services_in_registry
+        )
+
+
+async def _ensure_published_services_are_accessible(connection: SAConnection) -> None:
+    # Rationale: if a project template was published, its services must be available to everyone.
```

Comment:
I would challenge that rationale:  the mechanism should be the other way around. The access-rights to a study is determined by the most restrictive access-right of its components. 

Assume you have a user that creates a study and adds a component with access-rights defined by the component author (e.g. can only be executed by group X). Right now, he has the ability to publish the entire study and disregard the access-rights set by the author of the component.

Reply:
ok I will rename the function. We are talking here about published templates (the ones accessible from the portal). These ones must have accessible services.